drizzle-orm/src/mysql-core/dialect.ts

```diff
@@ -334,6 +335,54 @@ export class MySqlDialect {
 		return sql`${withSql}select${distinctSql} ${selection} from ${tableSql}${joinsSql}${whereSql}${groupBySql}${havingSql}${orderBySql}${limitSql}${offsetSql}${lockingClausesSql}`;
 	}
 
+	buildSetOperationQuery({
+		operator,
+		isAll,
+		leftSelect,
+		rightSelect,
+		limit,
+		orderBy,
+		offset,
+	}: MySqlSetOperatorConfig): SQL {
+		const leftChunk = sql`(${leftSelect.getSQL()}) `;
+		const rightChunk = sql`(${rightSelect.getSQL()})`;
+
+		let orderBySql;
+		if (orderBy && orderBy.length > 0) {
+			const orderByValues: SQL<unknown>[] = [];
+
+			// The next bit is necessary because the sql operator replaces ${table.column} with `table`.`column`
+			// which is invalid MySql syntax, Table from one of the SELECTs cannot be used in global ORDER clause
+			for (const orderByUnit of orderBy) {
+				if (is(orderByUnit, MySqlColumn)) {
+					orderByValues.push(sql.raw(orderByUnit.name));
+				} else if (is(orderByUnit, SQL)) {
+					for (let i = 0; i < orderByUnit.queryChunks.length; i++) {
```

Comment:
Shouldn't this be recursive? SQL's chunks are often SQL instances themselves.

Reply:
In all the manual and automatic testing I did, this was enough, I couldn't come up with a case where making this recursive was necessary.